dataplane/cmd/redbeams.go

```diff
@@ -15,32 +15,92 @@ import (
 func init() {
 	DataPlaneCommands = append(DataPlaneCommands, cli.Command{
 		Name:  "redbeams",
-		Usage: "manage RBDMS instances and databases",
+		Usage: "manage relational databases and database servers",
 		Subcommands: []cli.Command{
 			{
 				Name:  "dbserver",
-				Usage: "manage redbeams instances",
+				Usage: "manage redbeams database servers",
 				Subcommands: []cli.Command{
 					{
 						Name:   "list",
-						Usage:  "list all RBDMS instances",
+						Usage:  "list all database servers",
 						Before: cf.CheckConfigAndCommandFlagsWithoutWorkspace,
-						Flags:  fl.NewFlagBuilder().AddFlags(fl.FlEnvironmentCrn, fl.FlWaitOptional).AddAuthenticationFlags().Build(),
-						Action: redbeams.ListRBDMSInstances,
+						Flags:  fl.NewFlagBuilder().AddFlags(fl.FlEnvironmentCrn).AddAuthenticationFlags().AddOutputFlag().Build(),
+						Action: redbeams.ListDatabaseServers,
 						BashComplete: func(c *cli.Context) {
-							for _, f := range fl.NewFlagBuilder().AddFlags(fl.FlEnvironmentCrn, fl.FlWaitOptional).AddAuthenticationFlags().Build() {
+							for _, f := range fl.NewFlagBuilder().AddFlags(fl.FlEnvironmentCrn).AddAuthenticationFlags().AddOutputFlag().Build() {
+								fl.PrintFlagCompletion(f)
+							}
+						},
+					},
+					{
+						Name:   "get",
+						Usage:  "get a database server by CRN",
+						Before: cf.CheckConfigAndCommandFlagsWithoutWorkspace,
+						Flags:  fl.NewFlagBuilder().AddFlags(fl.FlCrn).AddAuthenticationFlags().AddOutputFlag().Build(),
+						Action: redbeams.GetDatabaseServer,
+						BashComplete: func(c *cli.Context) {
+							for _, f := range fl.NewFlagBuilder().AddResourceDefaultFlags().AddFlags(fl.FlCrn).AddAuthenticationFlags().AddOutputFlag().Build() {
+								fl.PrintFlagCompletion(f)
+							}
+						},
+					},
+					{
```

Comment:
By-name: good question if we do need this. Currently the name is a second class citizen - since it is not given by the user it does not matter to him if the identifier is a crn or a generated name beginning with dbstck. 
Now I would just delete (or maybe hide) the byName method and keep the crn one.
On the other hand I would collapse the describe + describe-by-name into one menu item, where there is a --crn and --name optional flag, or have a submenu below the describe:
+ dbserver
     ++ describe
          +++ by-crn
          +++ by-name

Reply:
I think using the naming convention of the other commands is a good idea.
The typical naming of commands "create, delete, list, describe".
So I think the best naming would be:
1) dp redbeams dbserver describe (--name ... || --crn)
2) dp redbeams dbserver describe by-name --name 
The second solution is a bit repetitive and long, so I would go with the first one.

Reply:
I'll try to implement option 1. It'd require either just --crn, or just --name and --environment-crn, but not both sets.

Maybe someday the redbeams API won't even support describe by name, and this will be much simpler.

Reply:
I have option 1 working locally, but I don't know if you will like how it works. The underlying cli library we use doesn't seem to support mutually exclusive sets of flags, so I implemented the flag checks on top of that library.

What I'll do is leave the separate commands here for this PR, but then I will submit a new PR just with the changes to combine the describe (and getstatus) commands as discussed here, and see if you are OK with how I implemented it.